build-logic/src/main/kotlin/PublicationPlugin.kt

```diff
@@ -26,122 +20,35 @@ abstract class PublicationPlugin : Plugin<Project> {
-                        version = version.toString()
-                        artifactId = localGradleProperty("POM_ARTIFACT_ID").get()
-
-                        artifact(sourcePlaceholderJar)
-                        artifact(docsPlaceholderJar)
-
-                        pom {
-                            name.set(localGradleProperty("POM_NAME"))
-                            description.set(providers.gradleProperty("POM_DESCRIPTION"))
-                            url.set(providers.gradleProperty("POM_URL"))
-                            licenses {
-                                license {
-                                    name.set(providers.gradleProperty("POM_LICENSE_NAME"))
-                                    url.set(providers.gradleProperty("POM_LICENSE_URL"))
-                                    distribution.set("repo")
-                                }
-                            }
-                            developers {
-                                developer {
-                                    id.set(providers.gradleProperty("POM_DEVELOPER_ID"))
-                                    name.set(providers.gradleProperty("POM_DEVELOPER_NAME"))
-                                }
-                            }
-                            scm {
-                                url.set(providers.gradleProperty("POM_SCM_URL"))
-                                connection.set(providers.gradleProperty("POM_SCM_CONNECTION"))
-                                developerConnection.set(providers.gradleProperty("POM_SCM_DEV_CONNECTION"))
-                            }
+            extensions.configure<MavenPublishBaseExtension> {
+                publishToMavenCentral()
+                coordinates(
+                    groupId = localGradleProperty("POM_GROUP_ID").get(),
+                    artifactId = localGradleProperty("POM_ARTIFACT_ID").get(),
+                    version = version.toString(),
+                )
+
+                // Pass `-PuseGpgCmd` to sign artifacts using local gpg agent
+                val useGpgCmd = localGradleProperty("useGpgCmd").orNull
+
+                val signingKeyId = localGradleProperty("signingKeyId").orNull?.takeIf { it.isNotEmpty() }
+                val signingKey = localGradleProperty("signingKey").orNull?.takeIf { it.isNotEmpty() }
+                val signingPassword = localGradleProperty("signingKeyPassword").orNull?.takeIf { it.isNotEmpty() }
+                val signingConfigPresent = signingKeyId != null && signingKey != null && signingPassword != null
+
+                // Avoid setting empty strings as signing keys. This avoids breaking the build when PR is opened from a fork.
+                // Also, due to https://github.com/gradle/gradle/issues/18477 signing tasks try to prematurely access the signatory.
+                // This also improves error messages if something's misconfigured
+                if (signingConfigPresent || useGpgCmd != null) {
```

Comment:
Previously using the gpg agent was possible only by uncommending piece of code. My proposal now is to expose extra parameter that allows people who don't have access to signing credentials to build signed artifacts locally 
(`./gradlew publishToMavenLocal -PuseGpgCmd`)